lib/transports/events-transport.js

```diff
@@ -11,13 +11,25 @@ class EventsTransport {
         this.logger = logger;
     }
 
-    async publish(stream, streamId, eventType, data, eventId, asOf) {
+    publish(stream, streamId, eventType, data, eventId, asOf) {
 
         this.logger.info({ msg: 'publishing event', stream, streamId, eventType, eventId, asOf });
 
-        const headers = { stream, streamId, eventType, eventId, asOf };
-
-        await this.publisher.publish(data, { key: `${stream}.${eventType}`, headers });
+        const options = { key: `${stream}.${eventType}` };
+
+        if (this.metadata) {
+            options.headers = this.metadata;
```

Comment:
nit: you might get away with ditching the `if` and setting the `options.headers` directly, eg

```suggestion
        options.headers = this.metadata || null;
```

Reply:
Done!